src/implementations/vanilla/components/date-block/date-block.scss

```diff
@@ -0,0 +1,186 @@
+$year-background-ongoing: map.get(color.$color, 'grey-10');
+$year-text-color-ongoing: map.get(color.$color, 'grey');
+$background-cancelled: map.get(color.$color, 'grey-10');
+$text-color-cancelled: map.get(color.$color, 'white');
+$separator-color-cancelled: map.get(color.$color, 'grey-25');
+$year-background-cancelled: map.get(color.$color, 'grey-10');
+$year-text-color-cancelled: map.get(color.$color, 'grey');
+$background-past: map.get(color.$color, 'white');
+$text-color-past: map.get(color.$color, 'grey');
+$separator-color-past: map.get(color.$color, 'grey-50');
+$year-background-past: map.get(color.$color, 'white');
+$year-text-color-past: map.get(color.$color, 'grey');
+$width-mobile: 5rem;
+$width-desktop: 6.3rem;
+$circle-width: 0.6rem;
+
+.ecl-date-block {
+  background-color: $background-default;
+  box-sizing: border-box;
+  display: inline-flex;
+  flex-direction: column;
+  margin: 0;
+  overflow: hidden;
+  position: relative;
+  text-align: center;
+  width: $width-mobile;
+
+  &::before {
+    background-color: map.get(color.$color, 'white');
+    border-radius: 50%;
+    box-sizing: content-box;
+    content: '';
+    display: block;
+    height: $circle-width;
+    left: map.get(spacing.$spacing, 's');
+    position: absolute;
+    top: -#{$circle-width / 2};
+    width: $circle-width;
+  }
+
+  &::after {
+    background-color: map.get(color.$color, 'white');
+    border-radius: 50%;
+    box-sizing: content-box;
+    content: '';
+    display: block;
+    height: $circle-width;
+    position: absolute;
+    right: map.get(spacing.$spacing, 's');
+    top: -#{$circle-width / 2};
+    width: $circle-width;
+  }
+}
+
+.ecl-date-block__day {
+  color: $text-color-default;
+  font: map.get(typography.$font, '2xl');
+  font-weight: map.get(typography.$font-weight, 'bold');
+  padding-left: map.get(typography.$font, 'xs');
+  padding-right: map.get(spacing.$spacing, '2xs');
+  padding-top: map.get(spacing.$spacing, 'xs');
+}
+
+.ecl-date-block__month {
+  color: $text-color-default;
+  font: map.get(typography.$font, 'm');
+  padding-bottom: map.get(spacing.$spacing, '2xs');
+  padding-left: map.get(spacing.$spacing, '2xs');
+  padding-right: map.get(spacing.$spacing, '2xs');
+  text-decoration: none !important;
+  text-transform: uppercase;
+}
+
+.ecl-date-block__year {
+  background-color: $year-background-default;
+  border-top: 1px solid $separator-color-default;
+  color: $year-text-color-default;
+  font: map.get(typography.$font, 's');
+  padding: map.get(spacing.$spacing, '2xs');
+}
+
+/* stylelint-disable-next-line order/order */
+@include breakpoints.media-breakpoint-up('lg') {
+  .ecl-date-block {
+    width: $width-desktop;
+  }
+
+  .ecl-date-block__month {
+    padding-bottom: map.get(spacing.$spacing, 'xs');
+  }
+
+  .ecl-date-block__year {
+    padding-bottom: map.get(spacing.$spacing, 'xs');
+    padding-top: map.get(spacing.$spacing, 'xs');
+  }
+
+  .ecl-date-block__daytime {
```

Comment:
where does this css come from? I don't see it in the current implementation

Reply:
Ah, this is for the "sr-only"  span, in the current implementation is using an utility class, i copied the css properties of the "ecl-u-sr-only" into this selector that i added, not only because we don't have utilities yet, but also because i believe those should not be used in the component markup.

Reply:
I see... I agree that components should not use utilities directly. I checked quickly, there are only a few components in this case, and we will update them when migrating.
But we should not use `!important` here (we shouldn't have any override issue inside a component)